src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -131,6 +132,7 @@ public static class WebScriptHostBuilderExtension
 
                     // Hosted services
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FileMonitoringService>());
+                    services.AddSingleton<IFileMonitoringService, FileMonitoringService>(serviceProvider => (FileMonitoringService)serviceProvider.GetServices<IHostedService>().FirstOrDefault(x => x is FileMonitoringService));
```

Comment:
The resolution here doesn't have to iterate over the hosted services. You can simply retrieve the registered `FileMonitoringService` instance in both cases (otherwise, we're just pushing the lookup cost up)

Reply:
I suppose we you are talking about the code:

```
services.AddSingleton<IFileMonitoringService, FileMonitoringService>();
services.TryAddSingleton<IHostedService>(p => (IHostedService)p.GetService<IFileMonitoringService>());
```
I this case we need to remove:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs#L48

Are we ok with that?

Reply:
We can adjust the validation as needed. The goal here is to register the `FileMonitoringService` as an `IFileMonitoringService` and that same instance as a hosted service, resolving the registered instance directly.